TelenorNBIoT.cpp

```diff
@@ -475,12 +470,24 @@ int TelenorNBIoT::parseErrorCode(const char *line)
 
 int TelenorNBIoT::splitFields(char *line, char **fields)
 {
-    int found = 1;
     fields[0] = line;
+    int found = 1;
     for(char *p = line; *p != '\0'; p++) {
-        if (*p == ',')
+        if (*p == '"')
+        {
+            // replace double-quote with null-terminator
+            *p = '\0';
+            
+            // if opening quote - move pointer one ahead
+            int current = found-1;
```

Comment:
If it's the opening quote then `found == 1`, so you can replace `current` with `0`.
Or even simpler:  check for `line[0] == '"'` before the loop 😄

Reply:
Or _even_ simpler:  do the unquoting in a different function, to completely separate the concerns.

So the caller would do something like this:
```
line = unquote(line)
found = splitFields(line, fields)
```

Reply:
Yep, my misunderstanding.